src/librustc_borrowck/diagnostics.rs

```diff
@@ -966,7 +972,7 @@ fn main() {
     };
     let borrowed = &mut cave;
 
-    borrowed.knight.nothing_is_true(); // E0507
+//  borrowed.knight.nothing_is_true(); // E0507
     mem::replace(&mut borrowed.knight, TheDarkKnight).nothing_is_true(); // ok!
 }
```

Comment:
Separate the success and failure cases to catch any regressions. For the second block it'll need a bunch of duplicated hidden code :-/